lib/internal/fetch/headers.js

```diff
@@ -0,0 +1,212 @@
+'use strict';
+
+const {
+  codes: { ERR_INVALID_ARG_VALUE },
+} = require('internal/errors');
+
+const {
+  ObjectEntries,
+  ArrayIsArray,
+  StringPrototypeToLocaleLowerCase,
+  StringPrototypeReplace,
+  Symbol,
+  SymbolIterator,
+  Array,
+} = primordials;
+
+const { validateObject } = require('internal/validators');
+
+const { isBoxedPrimitive } = require('internal/util/types');
+
+const { validateHeaderName, validateHeaderValue } = require('_http_outgoing');
+
+const { Buffer } = require('buffer');
+const console = require('console');
+
+const kHeadersList = Symbol('headers list');
+
+/**
+ * This algorithm is based off of
+ * https://www.tbray.org/ongoing/When/200x/2003/03/22/Binary
+ * It only operates on the even indexes of the array (the header names) by only
```

Comment:
```suggestion
 * It only operates on the even indices of the array (the header names) by only
```

Reply:
We follow the Microsoft Style Guide and [so `indexes` is the way to go here](https://docs.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/i/index-indexes-indices).